contracts/verifiers/UniversalVerifierMultiQuery.sol

```diff
@@ -0,0 +1,971 @@
+    struct AuthProof {
+        bool isVerified;
+        mapping(string key => uint256 inputValue) storageFields;
+        string validatorVersion;
+        uint256 blockTimestamp;
+    }
+
+    //TODO: custom errors to save some gas.
+
+    /**
+     * @dev Response. Structure for response.
+     * @param requestId Request id of the request.
+     * @param proof proof to verify.
+     * @param metadata Metadata of the request.
+     */
+    struct Response {
+        uint256 requestId;
+        bytes proof;
+        bytes metadata;
+    }
+
+    struct AuthType {
+        string authType;
+        IAuthValidator validator;
+        bytes params;
+    }
+
+    struct AuthTypeData {
+        IAuthValidator validator;
+        bytes params;
+        bool isActive;
+    }
+    struct AuthResponse {
+        string authType; //zkp-auth-v2, zkp-auth-v3, etc. will deside later
+        bytes proof;
+    }
+
+    struct RequestProofStatus {
+        uint256 requestId;
+        bool isVerified;
+        string validatorVersion;
+        uint256 timestamp;
+    }
+    struct AuthProofStatus {
+        string authType;
+        bool isVerified;
+        string validatorVersion;
+        uint256 timestamp;
+    }
+
+    struct GroupedResponses {
+        uint256 groupId;
+        Response[] responses;
+    }
+
+    struct UserAddressToIdInfo {
+        uint256 userID;
+        uint256 timestamp;
+    }
+
+    struct UserIdToAddressInfo {
+        address userAddress;
+        uint256 timestamp;
+    }
+
+    /**
+     * @dev Query. Structure for query.
+     * @param queryId Query id.
+     * @param requestIds Request ids for this multi query (without groupId. Single requests).
+     * @param groupIds Group ids for this multi query (all the requests included in the group. Grouped requests).
+     * @param metadata Metadata for the query. Empty in first version.
+     */
+    struct Query {
+        uint256 queryId;
+        uint256[] requestIds;
+        uint256[] groupIds;
+        bytes metadata;
+    }
+
+    /// @custom:storage-location erc7201:iden3.storage.UniversalVerifierMultiQuery
+    struct UniversalVerifierMultiQueryStorage {
+        // Information about requests
+        // solhint-disable-next-line
+        mapping(uint256 requestId => mapping(uint256 userID => Proof)) _proofs;
+        mapping(uint256 requestId => RequestData) _requests;
+        uint256[] _requestIds;
+        mapping(uint256 groupId => uint256[] requestIds) _groupedRequests;
+        uint256[] _groupIds;
+        IState _state;
+        // Information about queries
+        mapping(uint256 queryId => Query) _queries;
+        uint256[] _queryIds;
+        // Information linked between users and their addresses
+        mapping(address userAddress => uint256 userID) _user_address_to_id;
+        mapping(uint256 userID => address userAddress) _id_to_user_address;
+        mapping(uint256 userID => mapping(address userAddress => uint256 timestamp)) _user_auth_timestamp;
```

Comment:
The other mapping is missing here to support many-to-many
`mapping(address userAddress => mapping(uint256 userID => uint256 timestamp)) _auth_user_timestamp;`

Reply:
As discussed the comment is not relevant. Closing the discussion...